File: 1_BalPar/BalPar2/main.js
var antlr4 = require('antlr4');
var BalPar2Lexer = require('./BalPar2Lexer').BalPar2Lexer;
var BalPar2Parser = require('./BalPar2Parser').BalPar2Parser;

var input = "(()()())"
var chars = new antlr4.InputStream(input);

// Lexer step
var lexer = new BalPar2Lexer(chars);
var tokens  = new antlr4.CommonTokenStream(lexer);

// Print the tokens readed
tokens.fill();
tokens.getTokens(0, input.length).forEach(token =>
    console.log(token.toString())
)

// Parser step
var parser = new BalPar2Parser(tokens);
parser.buildParseTrees = true;
var tree = parser.parse_all();

// Print the parsing tree
if(tree.parser._syntaxErrors == 0){
    console.log("Parsing result = SUCCESS")
    console.log(tree.toStringTree(null, parser))
}

File: 1_BalPar/BalPar2/Main.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;
import java.io.FileInputStream;
import java.io.IOException;

public class Main {
  public static void main(String args[]) throws IOException{
    // Leggere input da file
    if(args.length == 0){
      System.out.println("Usare: 'java Main test'");
      return;
    }
    FileInputStream input_stream = new FileInputStream(args[0]);
    CharStream input = CharStreams.fromStream(input_stream); 
    
    // Creare l'oggetto Lexer
    BalPar2Lexer lexer = new BalPar2Lexer(input);
    CommonTokenStream tokens = new CommonTokenStream(lexer); 

    // Print token letti
    tokens.fill();
    for(Token token : tokens.getTokens()){
        System.out.println(token.toString());
    }

    // Creare il parser ed eseguire il parsing
    BalPar2Parser parser = new BalPar2Parser(tokens);
    ParseTree tree = parser.parse_all();

    // Risultato del parsing e print del parse tree
    if(parser.getNumberOfSyntaxErrors() == 0){
      System.out.println("Parsing result = SUCCESS");
      // Print the parsing tree
      System.out.println(tree.toStringTree(parser));
    }
  }
}